plugins/services/src/js/reducers/serviceForm/JSONReducers/HealthChecks.js

```diff
@@ -8,23 +8,38 @@ import {
   MESOS_HTTPS
 } from "../../../constants/HealthCheckProtocols";
 
-function mapHealthChecks(item) {
-  const newItem = Util.omit(item, ["path", "command", "protocol", "https"]);
-
-  if (item.protocol != null) {
-    newItem.protocol = item.protocol;
-    if (item.protocol.toUpperCase() === COMMAND && item.command != null) {
-      newItem.command = {
-        value: item.command
-      };
-    }
+function getMapHealthChecks(docker) {
```

Comment:
🎨   Is there any better way to pass `docker`? How about binding the function (e.g.,  `mapHealthChecks.bind(null, docker)`) or creating a support structure that includes the needed information?

Reply:
Well how is binding the variable and in that way creating a new function a better way then currying the docker?

Reply:
I'd argue that changing the function signature to  `mapHealthChecks(docker, item)` and then partially applying `docker` is easier to reason about. It's also using generic operations instead of creating a very specific `getMapHealthChecks` function that returns a _bound_ version of `mapHealthChecks`. Anyway, as indicated this is a purely stylistic comment and by no means blocks this PR.

Reply:
I'm ok with either of those, but how about renaming docker argument to be `runtime`? We're using it only once here `docker.type === "DOCKER"` to become `runtime === "DOCKER"`.

*Side note*: it's all that pesky functional programming. In OOP I'd just create two mappers and decided to use one or another upfront without obscure functions that return function :trollface:

Reply:
Well in OOP you would consider to have two classes which will do this and a factory which would return either a docker runtime mapper or a not docker runtime mapper based on the input. I guess it is far more code to maintain with a slim curry.